src/openforms/js/components/admin/form_design/variables/ServiceFetchConfigurationPicker.js

```diff
@@ -0,0 +1,94 @@
+import {useFormik} from 'formik';
+import PropTypes from 'prop-types';
+import React, {useContext, useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import Field from 'components/admin/forms/Field';
+import FormRow from 'components/admin/forms/FormRow';
+import Select from 'components/admin/forms/Select';
+
+import {FormLogicContext} from './../Context';
+import ServiceFetchConfigurationForm from './ServiceFetchConfigurationForm';
+
+const ServiceFetchConfigurationPicker = ({data = {}, onChange, onFormSave}) => {
+  const formLogicContext = useContext(FormLogicContext);
+
+  const [selectExisting, setSelectExisting] = useState(false);
+  const [selectedServiceFetchConfig, setSelectedServiceFetchConfig] = useState(null);
+
+  const serviceFetchConfigurationChoices = formLogicContext.serviceFetchConfigurations.map(
+    fetchConfig => {
+      return [
+        fetchConfig.url,
+        fetchConfig.name || `${fetchConfig.method} ${fetchConfig.path} (${fetchConfig.service})`,
+      ];
+    }
+  );
+
+  const formik = useFormik({
+    initialValues: {
+      method: 'GET',
+      service: '',
+      path: '',
+      queryParams: [['', ['']]],
+      headers: [['', '']],
+      body: '',
+      dataMappingType: '',
+      mappingExpression: '',
+    },
+    onSubmit: (values, {setSubmitting}) => {
+      alert(JSON.stringify(values, null, 2));
+    },
+  });
+
+  return (
+    <div className="servicefetchconfiguration-picker">
+      <div className="servicefetchconfiguration-select">
+        <FormRow>
+          <Field
+            name={'fetchConfiguration.existing'}
+            label={
+              <FormattedMessage
+                defaultMessage="Choose existing configuration"
+                description="Service fetch configuration modal select existing field label"
+              />
+            }
+          >
+            <Select
+              name="fetchConfiguration.existing"
+              choices={serviceFetchConfigurationChoices}
+              value={selectedServiceFetchConfig}
+              onChange={event => {
+                onChange(event);
+                setSelectedServiceFetchConfig(event.target.value);
+                formik.setValues(
+                  formLogicContext.serviceFetchConfigurations.find(
+                    element => element.url === event.target.value
+                  ) || formik.initialValues
+                );
+
+                setSelectExisting(!!event.target.value);
+              }}
+              allowBlank
+            />
+          </Field>
+        </FormRow>
+      </div>
+
+      <div className="servicefetchconfiguration-form">
```

Comment:
this should probably be a proper `<form onSubmit={formik.handleSubmit)>` tag :)

Reply:
I'd put it close to where you'll actually handle the submit event, so probably here. That does also improve composabililty (because you can't put form tags inside other form tags).